plugin-gradle/README.md

```diff
@@ -774,6 +775,25 @@ spotless {
 
 If you'd like to create a one-off Spotless task outside of the `check`/`apply` framework, see [`FormatExtension.createIndependentApplyTask`](https://javadoc.io/static/com.diffplug.spotless/spotless-plugin-gradle/5.5.2/com/diffplug/gradle/spotless/FormatExtension.html#createIndependentApplyTask-java.lang.String-).
 
+## Inception (languages within languages within...)
+
+In very rare cases, you might want to format e.g. javascript which is written inside JSP templates, or maybe java within a markdown file, or something wacky like that.  You can specify hunks within a file using either open/close tags or a regex with a single capturing group, and then specify rules within it, like so.  See [javadoc TODO](https://javadoc.io/static/com.diffplug.spotless/spotless-plugin-gradle/5.5.1/com/diffplug/gradle/spotless/FormatExtension.html#target-java.lang.Object...-) for more details.
+
+```gradle
+import com.diffplug.gradle.spotless.JavaExtension
+
+spotless {
+  format 'templates', {
+    target 'src/templates/**/*.foo.html'
+    prettier().config(['parser': 'html'])
+    withinBlocks 'javascript block', '<script>', '</script>', {
+      prettier().config(['parser': 'javascript'])
+    }
+    withinBlocksRegex 'single-line @(java-expresion)', '@\\((.*?)\\)', JavaExtension, {
```

Comment:
I'm not clear on what the regex `@\\((.*?)\\)` matches. Should we add a comment or replace it with the example in the Javadoc of `FormatExtension.withinBlocks`?

Reply:
In some java-html templating systems (I'm thinking of rocker and play framework 2), you can pass java variables to templates, and render any arbitrary expression with `@(someExpression)`, e.g. `@(i+1)`.  I'm happy for this regex to get changed to anything else, I personally always find them difficult to read.  IMO, the intention is just to show "pass a string and it will be used as a regex".

Reply:
Ahh okay, that makes a lot of sense! Goes to show that I'm not familiar with that class of templating systems. 😛

My opinion is that the example in the doc for `FormatExtension.withinBlocks` is easier to understand, since it doesn't rely on an understanding of a certain type of templating system, which I think it makes the intention clearer. But I'm happy to defer to you on this.

Reply:
I agree with you, but not enough to dig up the javadoc, make the change, and push it.  If you want to do it, by all means :)

Reply:
Likewise I'm not fussed enough to want to make the change myself, so let's leave it, it's good enough. :)